cmd/lifecycle-util.go

```diff
@@ -0,0 +1,277 @@
+	expiryDatesLabelKey     string = "Expiry-Dates"
+	transitionDatesLabelKey string = "Transition-Date"
+	transitionDaysLabelKey  string = "Transition-Days"
+)
+
+const (
+	expiryDatesLabelFlag string = "Expiry-Date"
+	expiryDaysLabelFlag  string = "Expiry-Days"
+)
+
+const (
+	fieldMainHeader         string = "Main-Heading"
+	fieldThemeHeader        string = "Row-Header"
+	fieldThemeRow           string = "Row-Normal"
+	fieldThemeTick          string = "Row-Tick"
+	fieldThemeExpiry        string = "Row-Expiry"
+	fieldThemeResultSuccess string = "SucessOp"
+	fieldThemeResultFailure string = "FailureOp"
+)
+
+const (
+	tickCell      string = "\u2713 "
+	crossTickCell string = "\u2717 "
+	blankCell     string = "  "
+)
+
+const defaultILMDateFormat string = "2006-01-02"
+
+const (
+	tagSeperator    string = ","
+	keyValSeperator string = ":"
+	tableSeperator  string = "|"
+)
+
+type tableCellInfo struct {
+	label       string
+	multLabels  []string
+	labelKey    string
+	fieldTheme  string
+	columnWidth int
+	align       int
+}
+
+type showDetails struct {
+	allAvailable bool
+	expiry       bool
+	transition   bool
+	json         bool
+	minimum      bool
+}
+
+func getCentered(label string, maxLen int) string {
+	const toPadWith string = " "
+	lblLth := len(label)
+	if lblLth > 1 && lblLth%2 != 0 {
+		lblLth++
+	} else if lblLth == 1 {
+		lblLth = 2
+	}
+	length := (float64(maxLen - lblLth)) / float64(2)
+	rptLth := (int)(math.Floor(length / float64(len(toPadWith))))
+	leftRptLth := rptLth
+	rightRptLth := rptLth
+	if rptLth <= 0 {
+		leftRptLth = 1
+		rightRptLth = 0
+	}
+	output := strings.Repeat(toPadWith, leftRptLth) + label + strings.Repeat(toPadWith, rightRptLth)
+	return output
+}
+
+func getLeftAlgined(label string, maxLen int) string {
+	const toPadWith string = " "
+	lblLth := len(label)
+	length := maxLen - lblLth
+	if length <= 0 {
+		return label
+	}
+	output := strings.Repeat(toPadWith, 1) + label + strings.Repeat(toPadWith, length-1)
+	return output
+}
+
+func getRightAligned(label string, maxLen int) string {
+	const toPadWith string = " "
+	lblLth := len(label)
+	length := maxLen - lblLth
+	if length <= 0 {
+		return label
+	}
+	output := strings.Repeat(toPadWith, length) + label
+	return output
+}
+
+func getBucketNameFromURL(urlStr string) string {
+	clientURL := newClientURL(urlStr)
+	bucketName := splitStr(clientURL.String(), "/", 3)[1]
+	return bucketName
+}
+
+func getHostCfgFromURL(urlStr string) *hostConfigV9 {
+	clientURL := newClientURL(urlStr)
+	alias := splitStr(clientURL.String(), "/", 3)[0]
+	return mustGetHostConfig(alias)
+}
+
+func getParsedHostURL(urlStr string) (*url.URL, *probe.Error) {
+	hostCfg := getHostCfgFromURL(urlStr)
+	if hostCfg == nil {
+		return nil, errInvalidURL(urlStr)
+	}
+	hostURLParse, err := url.Parse(hostCfg.URL)
+	if err != nil {
+		return nil, probe.NewError(err)
+	}
+	return hostURLParse, nil
+}
+
+func getMinioClient(urlStr string) (*minio.Client, *probe.Error) {
```

Comment:
This type of code is unusual - what is the difficulty in using the client-s3.go? @BigUstad  - this makes assumptions only with v4 support and overrides `mc config` requests to use signature v2.